src/app/Dashboard/AddCard.tsx

```diff
@@ -157,3 +226,166 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
     </>
   );
 };
+
+interface PropsConfigFormProps {
+  cardTitle: string;
+  controls: PropControl[];
+  config: any;
+  onChange: ({}) => void;
+}
+
+const PropsConfigForm = (props: PropsConfigFormProps) => {
+  const handleChange = React.useCallback(
+    (k) => (e) => {
+      const copy = { ...props.config };
+      copy[k] = e;
+      props.onChange(copy);
+    },
+    [props.config, props.onChange]
+  );
+
+  const handleNumeric = React.useCallback(
+    (k) => (e) => {
+      const value = (e.target as HTMLInputElement).value;
+      const copy = { ...props.config };
+      copy[k] = value;
+      props.onChange(copy);
+    },
+    [props.config, props.onChange]
+  );
+
+  const handleNumericStep = React.useCallback(
+    (k, v) => (e) => {
+      const copy = { ...props.config };
+      copy[k] = props.config[k] + v;
+      props.onChange(copy);
+    },
+    [props.config, props.onChange]
+  );
+
+  const createControl = React.useCallback(
+    (ctrl: PropControl): JSX.Element => {
+      let input: JSX.Element;
+      switch (ctrl.kind) {
+        case 'boolean':
+          input = (
+            <Switch label="On" labelOff="Off" isChecked={props.config[ctrl.key]} onChange={handleChange(ctrl.key)} />
+          );
+          break;
+        case 'number':
+          input = (
+            <NumberInput
+              inputName={ctrl.name}
+              inputAriaLabel={`${ctrl.name} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleNumeric(ctrl.key)}
+              onPlus={handleNumericStep(ctrl.key, 1)}
+              onMinus={handleNumericStep(ctrl.key, -1)}
+            />
+          );
+          break;
+        case 'string':
+          input = (
+            <TextInput
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'text':
+          input = (
+            <TextArea
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'select':
+          input = (
+            <SelectControl handleChange={handleChange(ctrl.key)} config={props.config[ctrl.key]} control={ctrl} />
```

Comment:
I think `config` prop should just be `props.config`? Right now, selecting something from menu does not render the selected option.